design-proposals/eim-decouple-microos-storage.md

```diff
@@ -0,0 +1,185 @@
+
+```mermaid
+sequenceDiagram
+%%{wrap}%%
+autonumber
+
+  participant user as User
+
+  participant en as Edge Node
+
+  box rgb(235,255,255) Edge Orchestrator
+
+      participant pa as Provisioning Nginx
+      participant om as Onboarding Manager
+      participant dkam as DKAM
+      participant ic as infra-config
+      participant inv as Inventory / API
+
+  end
+
+  box rgb(235,255,255) Edge Microvisor Toolkit Build System
+      participant emt-ci as EMT CI/CD Pipeline
+  end
+
+  participant rs as Release Service File Server
+
+  rect rgb(191, 223, 255)
+
+  note right of emt-ci: Day -1, Micro-EMT build process
+
+  emt-ci->>emt-ci: Build Micro-EMT with all required agents
+  emt-ci->>emt-ci: Sign generated Micro-EMT with the DER security key
+  emt-ci->>rs: Publish signed Micro-EMT image along with the DER security key 
+
+  end
+
+  ic->>dkam: Read orchestrator CA certificate and cluster-wide variables
+
+  dkam->>dkam: Generates and signs Micro-OS bundle
+
+  dkam->>pa: Store Micro-OS bundle on PV 
+
+  dkam->>dkam: Builds and signs iPXE EFI binary
+
+  dkam->>pa: Store signed iPXE EFI binary
+
+  user->>+rs: Download RS certificate & Micro-OS DER key
+  rs->>-user: [RS certificate, Micro-OS DER key]
+
+  user->>en: Load RS certificate & Micro-OS DER key to BIOS
+
+  note over en: EN powered on
+
+  en->>pa: Download iPXE binary
+  en->>en: EN boots into iPXE binary
+
+  en->>+rs: Download Micro-OS image
+  rs->>-en: [Micro-OS image]
+
+  en->>pa: Download the Micro-OS bundle
+
+  en->>en: Injects the Micro-OS bundle files to Micro-OS via initrd
+  
+  en->>en: Loads initramfs via initrd
+
+  en->>en: Boots into Micro-OS
+
+  note over user,rs: EN continues Onboarding and Provisioning
+```
+
+Steps 1-3: The CI/CD pipeline of Micro-EMT build the image and publishes it to the Release Service (S3 bucket), along with the signing key.
+The process is the same as currently used for the EMT image.
+
+Steps 4-8: At deployment time, DKAM reads the orchestrator CA certificate and cluster-wide variables,
+and generates two artifacts: 1) the signed Micro-OS bundle, and 2) the signed iPXE script that downloads 
+the Micro-OS image and the Micro-OS bundle.
+We assume that both artifacts are signed with the same key that is generated by DKAM.
+Note that iPXE includes standard public certificates (like Microsoft or AWS) so we don't need to embed
+these certificates, as long as it's not trusted by iPXE or a self-signed certificate that is used by a local storage.
+
+Steps 9-11: User needs to download a set of certificates and signing keys and upload them to BIOS. The set should include:
+- the orchestrator CA certificate (uploaded now)
+- the secure key used to sign iPXE and orchestrator bundle (uploaded now)
+- the secure key used to sign EMT (uploaded now) - only if EMT is used
+- the secure key used to sign Micro-OS image (new). This key is downloaded from the external storage.
```

Comment:
```suggestion
- the secure key used to sign Micro-OS image (uploaded now). This key is downloaded from the external storage (new).
```

Reply:
Are you thinking of the DER key that is currently used to sign EMT images? I'm not sure if we can use the same one.